service/prometheus/filter_test.go

```diff
@@ -0,0 +1,90 @@
+package prometheus
+
+import (
+	"reflect"
+	"testing"
+
+	"github.com/davecgh/go-spew/spew"
+	"k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// Test_Prometheus_FilterInvalidServices tests the FilterInvalidServices function.
+func Test_Prometheus_FilterInvalidServices(t *testing.T) {
+	tests := []struct {
+		services         []v1.Service
+		expectedServices []v1.Service
+	}{
+		// Test a service without a cluster annotation is filtered.
+		{
+			services: []v1.Service{
+				{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      "foo",
+						Namespace: "bar",
+					},
+				},
+			},
+			expectedServices: []v1.Service{},
+		},
+
+		// Test that two services without cluster annotations are filtered.
+		{
+			services: []v1.Service{
+				{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      "foo",
+						Namespace: "bar",
+					},
+				},
+
+				{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      "baz",
+						Namespace: "bar",
+					},
+				},
+			},
+			expectedServices: []v1.Service{},
+		},
+
+		// Test that a service with a cluster annotation is not filtered.
+		{
+			services: []v1.Service{
+				{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      "foo",
+						Namespace: "bar",
+						Annotations: map[string]string{
+							ClusterAnnotation: "xa5ly",
+						},
+					},
+				},
+			},
+			expectedServices: []v1.Service{
+				{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      "foo",
+						Namespace: "bar",
+						Annotations: map[string]string{
+							ClusterAnnotation: "xa5ly",
+						},
+					},
+				},
+			},
+		},
+	}
```

Comment:
A test case with multiple valid/partial valid services is missing.

Reply:
Yeah, I'm not super certain what to do in these cases yet. Will make sure it's addressed in a follow up.